server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -1000,7 +1004,7 @@ public Entry abort() {
             if (allQueued) {
                 return null;
             }
-            return new Entry(
+            return Entry.snapshot(
```

Comment:
This seems a little buggy in the case of clones where we could leak some of the blobs in the shard directories by turning a clone into a snapshot on abort. The only thing that changes here is that we "admit" that we're doing it by setting `source` to `null` but that's irrelevant for downstream logic. This is fortunately not so bad otherwise since the completed entry will still be removed from the CS and repo metadata just fine otherwise and the leaked blobs are referenced by at least one other snapshot already.
I'll clean this up in a follow-up though.